projects/uitdatabank/reference/entry.json

```diff
@@ -2614,6 +2614,79 @@
+        "responses": {
+          "204": {
+            "description": "No Content. The workflowStatus was successfully updated."
+          },
+          "400": {
+            "description": "Bad Request. Possible error types:\n\n* https://api.publiq.be/probs/body/missing\n* https://api.publiq.be/probs/body/invalid-syntax\n* https://api.publiq.be/probs/body/invalid-data",
+            "content": {
+              "application/problem+json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                },
+                "examples": {
+                  "Invalid data": {
+                    "value": {
+                      "type": "https://api.publiq.be/probs/body/invalid-data",
+                      "title": "Invalid body data",
+                      "status": 400,
+                      "schemaErrors": [
+                        {
+                          "error": "Missing required properties (workflowStatus)",
+                          "jsonPointer": "/"
+                        }
+                      ]
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/Unauthorized"
+          },
+          "403": {
+            "$ref": "#/components/responses/Forbidden"
+          },
+          "404": {
+            "$ref": "#/components/responses/NotFound"
+          }
+        },
+        "operationId": "event-workflow-status-put",
+        "description": "Updates the workflow status of an event. Possible statuses are:\n\n- `DRAFT`: The default status of new events. Events with this status do not appear in online calendars like <https://www.uitinvlaanderen.be> or the search on <https://www.uitdatabank.be>\n- `READY_FOR_VALIDATION`: This status means the event has been published, but not approved yet. Most online calendars will already show it, and it will appear in the search on <https://www.uitdatabank.be>\n- `APPROVED`: The event has been approved by a moderator. It will appear on all online calendars. You cannot set this status unless you have moderation permissions.\n- `REJECTED`: The event has been rejected by a moderator. It will not appear on any online calendars. You cannot set this status unless you have moderation permissions.\n- `DELETED`: The event has been deleted. It will not appear on any online calendars.",
+        "security": [
+          {
+            "USER_ACCESS_TOKEN": []
+          },
+          {
+            "CLIENT_ACCESS_TOKEN": []
+          }
+        ],
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "../models/event-workflowStatus-put.json"
+              },
+              "examples": {}
+            }
+          },
+          "description": "New workflowStatus to set on the event. Depending on the new workflowStatus, other properties may/must be set as well."
```

Comment:
Maybe define which properties must be set as well?

Reply:
I can add it, but it's also mentioned in the descriptions of the possible properties. I've tweaked the descriptions slightly to clarify it. See screenshot below.

![Screen Shot 2022-11-28 at 13 53 07](https://user-images.githubusercontent.com/959026/204282565-ca603021-a31c-495e-957a-7d18cacd9a3f.png)

Is this okay or do you think we should also mention it in the description of the complete body?

Reply:
For reference, you can preview the endpoint here (when logged in): https://docs.publiq.be/docs/uitdatabank/branches/uitdatabank%2FIII-5106-document-workflow-status-endpoints/4c00e3267c774-workflow-status-update